airbyte-webapp/src/components/base/SecretTextArea/SecretTextArea.tsx

```diff
@@ -0,0 +1,93 @@
+import { faEye } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+import { useMemo, useRef } from "react";
+import { FormattedMessage } from "react-intl";
+import { useToggle, useUpdateEffect } from "react-use";
+
+import { TextInputContainer, TextInputContainerProps } from "../TextInputContainer";
+import styles from "./SecretTextArea.module.scss";
+
+interface SecretTextAreaProps
+  extends Omit<TextInputContainerProps, "onFocus" | "onBlur">,
+    React.TextareaHTMLAttributes<HTMLTextAreaElement> {}
+
+export const SecretTextArea: React.VFC<SecretTextAreaProps> = ({
+  name,
+  disabled,
+  value,
+  onMouseUp,
+  onBlur,
+  error,
+  light,
+  ...textAreaProps
+}) => {
+  const hasValue = useMemo(() => !!value && String(value).trim().length > 0, [value]);
+  const [isContentVisible, toggleIsContentVisible] = useToggle(!hasValue);
+  const textAreaRef = useRef<HTMLTextAreaElement | null>(null);
+  const textAreaHeightRef = useRef<number>((textAreaProps.rows ?? 1) * 20 + 14);
+
+  useUpdateEffect(() => {
+    if (isContentVisible && textAreaRef.current) {
+      textAreaRef.current.focus();
+      const selectionStart = value ? String(value).length : 0;
+      textAreaRef.current.setSelectionRange(selectionStart, selectionStart);
+    }
+  }, [isContentVisible]);
+
+  return (
+    <TextInputContainer disabled={disabled} error={error} light={light}>
+      {isContentVisible ? (
+        <textarea
+          spellCheck={false}
+          {...textAreaProps}
+          className={classNames(styles.textarea, "fs-exclude", textAreaProps.className)}
+          name={name}
+          disabled={disabled}
+          ref={textAreaRef}
+          onMouseUp={(event) => {
+            textAreaHeightRef.current = textAreaRef.current?.offsetHeight ?? textAreaHeightRef.current;
+            onMouseUp?.(event);
+          }}
+          onBlur={(event) => {
+            textAreaHeightRef.current = textAreaRef.current?.offsetHeight ?? textAreaHeightRef.current;
+            if (hasValue) {
+              toggleIsContentVisible();
+            }
+            onBlur?.(event);
+          }}
+          style={{ height: textAreaHeightRef.current }}
+          value={value}
+          data-testid="textarea"
+        />
+      ) : (
+        <>
+          <button
+            type="button"
+            className={styles.toggleVisibilityButton}
+            onClick={() => {
+              toggleIsContentVisible();
+            }}
+            style={{
+              height: textAreaHeightRef.current,
+            }}
+            disabled={disabled}
+            data-testid="visibilityButton"
+          >
+            <FormattedMessage id="ui.secretTextArea.hidden" /> <FontAwesomeIcon icon={faEye} fixedWidth />
```

Comment:
:art: Minor nitpick (and I don't have any really strong feelings around this): It feels the icon might look a bit nicer in the front? I guess it's in the back because it's in the back for the 1 line password fields, but it feels more "right" there since it's right aligned, here it's just attached to the text, where I feel left might be the better option.

Reply:
I tried it and I also think it looks better in the front.